File: tests/text_io/text_io.java
package tests.text_io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class text_io {
    public static void main(String[] args) throws IOException {
        String load_path = "../../json/random" + args[0] + ".json";
        String store_path = "../../json/java_out_" + args[0] + ".txt";

        String content = Files.readString(Paths.get(load_path));
        Files.writeString(Paths.get(store_path), content);
    }
}

File: tests/text_io/text_io.cpp
#include <fstream>
#include <sstream>
#include <iostream>
using namespace std;

int main(int argc, char* argv[]) {
    string file_data;
    string load_path = "json/random" + string(argv[1]) + ".json";
    string store_path = "json/cpp_out_" + string(argv[1]) + ".txt";

    // load data
    ifstream read_file(load_path);
    stringstream sstr;
    read_file >> sstr.rdbuf();
    file_data = sstr.str();
    read_file.close();

    // cout << file_data << endl;

    // save data
    ofstream write_file(store_path);
    write_file << file_data;
    write_file.close();

    return 0;
}